app/components/goodcity/orders-package-block.js

```diff
@@ -0,0 +1,204 @@
+  }
+};
+
+const ACTIVE_ORDER_STATES = [
+  "submitted",
+  "processing",
+  "awaiting_dispatch",
+  "dispatching"
+];
+
+const FORCE_DISABLE = ["edit_quantity"];
+
+const CANCELLED = {};
+
+/**
+ * UI Component representing an orders_package
+ *
+ * --------------------------------------------------------
+ * | op_state  |   order_code         order_state |    <  |
+ * |           |                                  |  <    |
+ * | qty       |   organisation                   |    <  |
+ * --------------------------------------------------------
+ *
+ * It provides a drawer from the right side which is populated
+ * with the actions provided by the backend
+ *
+ * Clicking on an action will run it on the API side
+ *
+ * @property {Model} orderPkg the orders_package to display
+ *
+ * Usage:
+ *  {{goodcity/orders-package-block orderPkg=orderPkg }}
+ */
+export default Ember.Component.extend({
+  store: Ember.inject.service(),
+  messageBox: Ember.inject.service(),
+  i18n: Ember.inject.service(),
+  designationService: Ember.inject.service(),
+  actionRunner: Ember.computed.alias("designationService"),
+
+  init() {
+    this._super(...arguments);
+    this.set("paramGetters", {
+      order_id: "selectOrder",
+      quantity: "selectQuantity"
+    });
+  },
+
+  // --- PARAMS RESOLUTION
+
+  /**
+   * Prompts the user to select an order, and returns its id
+   *
+   * CANCELLED is returned if the user closes the UI
+   *
+   * @returns {Promise<String>}
+   */
+  async selectOrder() {
+    const order = await this.get("designationService").userPickOrder({
+      state: ACTIVE_ORDER_STATES.join(",")
+    });
+
+    return order ? order.get("id") : CANCELLED;
+  },
+
+  /**
+   * Opens a qty selection UI and resolves the promise with the desired
+   * quanntity
+   *
+   * CANCELLED is returned if the user closes the UI
+   *
+   * @todo
+   * @returns {Promise<number>}
+   */
+  async selectQuantity() {
+    throw new Error("Not implemennted");
+  },
+
+  /**
+   * Given an actionn name, returns an object with all the required properties
+   * to successfully
+   *
+   * CANCELLED if the user did not provide the
+   */
+  async fulfillParams(actionName) {
+    const paramNames = _.get(ACTIONS_SETTINGS, `${actionName}.params`, []);
+
+    let params = {};
+    for (let name of paramNames) {
+      const getterName = this.getWithDefault(`paramGetters.${name}`, "");
+      const val = await this[getterName]();
+
+      if (val === CANCELLED) {
+        return CANCELLED;
+      }
+      params[name] = val;
+    }
+    return params;
+  },
+
+  // --- EXECUTION
+
+  showSpinner() {
+    Ember.run(() => {
+      if (!this.loadingView) {
+        this.loadingView = Ember.getOwner(this)
+          .lookup("component:loading")
+          .append();
+      }
+    });
+  },
+
+  hideSpinner() {
+    Ember.run(() => {
+      if (this.loadingView) {
+        this.loadingView.destroy();
+        this.loadingView = null;
+      }
+    });
+  },
+
```

Comment:
Wouldn't it be better if we make common mixins which will have these uitlity for spinners, catching errors etc. Like similar we have in admin. We should import these mixins in root component, routes and controllers. Through which all components, controllers and routes should inherit.
Thoughts ?

Reply:
Yeah, we have one in Browse (I think), but I got lazy 😄 
I'll bring it to Stock

Reply:
oh Admin, not browse. Like you said